apis/v1alpha1/collector_webhook.go

```diff
@@ -29,11 +30,19 @@ import (
 	"github.com/open-telemetry/opentelemetry-operator/internal/config"
 	ta "github.com/open-telemetry/opentelemetry-operator/internal/manifests/targetallocator/adapters"
 	"github.com/open-telemetry/opentelemetry-operator/pkg/featuregate"
+	"github.com/open-telemetry/opentelemetry-operator/pkg/rbac"
 )
 
 var (
 	_ admission.CustomValidator = &CollectorWebhook{}
 	_ admission.CustomDefaulter = &CollectorWebhook{}
+
+	// targetAllocatorNamespaceRes is the resource attributes required for the newest target allocator capability
+	targetAllocatorNamespaceRes = &v1.ResourceAttributes{
+		Namespace: "",
+		Verb:      "list",
```

Comment:
Thanks @jaronoff97 for making this change! :) The Prometheus CR watcher requires watch permission as well, should we include that in these attributes as well?

Reply:
oh its both? sure! can do. I think i may introduce an intermediary structure so we can just list permissions.

Reply:
Sounds good! Thanks